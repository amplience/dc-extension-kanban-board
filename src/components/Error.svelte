<script lang="ts">
  export let reason: string = '';
</script>

<style lang="scss">
  .error {
    height: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
    text-align: center;
    font-size: 16px;
    color: #999;
    .message {
      max-width: 500px;
    }
  }
</style>

<div class="error">
  <div class="message">{reason}</div>
</div>
