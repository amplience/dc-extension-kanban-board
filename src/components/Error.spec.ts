import { render } from '@testing-library/svelte';
import Error from './Error.svelte';

describe('error component', () => {
  it('should render an error component with value', () => {
      const { container } = render(Error, {
        reason: 'REASON'
      });

      expect(container.firstChild).toMatchSnapshot();
  });

  it('should render an error component with default value', () => {
      const { container } = render(Error, {
      // no-op.
      });

      expect(container.firstChild).toMatchSnapshot();
  });
});