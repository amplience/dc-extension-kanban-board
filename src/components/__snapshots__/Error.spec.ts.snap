// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error component should render an error component with default value 1`] = `
<div>
  <div
    class="error svelte-q74y49"
  >
    <div
      class="message svelte-q74y49"
    >
      
    </div>
  </div>
</div>
`;

exports[`error component should render an error component with value 1`] = `
<div>
  <div
    class="error svelte-q74y49"
  >
    <div
      class="message svelte-q74y49"
    >
      REASON
    </div>
  </div>
</div>
`;
